neqo-http3/src/send_message.rs

```diff
@@ -166,7 +166,12 @@ impl Stream for SendMessage {
 }
 impl SendStream for SendMessage {
     fn send_data(&mut self, conn: &mut Connection, buf: &[u8]) -> Res<usize> {
-        qtrace!([self], "send_body: len={}", buf.len());
+        qtrace!([self], "send_data: len={}", buf.len());
+
+        if buf.is_empty() {
+            qerror!([self], "zero-length send_data");
+            return Err(Error::InvalidInput);
+        }
```

Comment:
Same as in `neqo_transport::SendStream::send_internal` which is called by this function anyways. Thus just moving the `return Err` up the stack, which allows the `else if available <= 2` to depend on `buf` not to be empty.

https://github.com/mozilla/neqo/blob/3e537097cbd6e75b861d0d859f9c5eb9c233a723/neqo-transport/src/send_stream.rs#L1253-L1256.